### PR TITLE
Mention "original page name" in mouse-over in Tree view

### PR DESCRIPTION
**Pull request by JorisDebonnet**
I am trying to display the "original page name" in the mouseover on SiteTree objects in the tree view (see also #234), but I don't manage to "Hello World!" yet. 

As far as I can see, the mouseover is generated in `LeftAndMain_TreeNode->forTemplate`, so I tried to create an extension for that class and overwrite the `forTemplate` method. That isn't working; what did I forget?

### CODE REVIEW

File: code/controller/TranslatableCMSTreeNodeExtension.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	}
+
+	/**
+	 * Overrides {@link LeftAndMain_TreeNode->forTemplate()} in order to show the original page name on mouseover.
+	 *
+	 * @todo Make it more flexible, because right now it's a simple copy-paste with changes
+	 *
+	 * @return String
+	 */
+	public function forTemplate() {
```
Comment (dhensby) [inline code/controller/TranslatableCMSTreeNodeExtension.php]: Couldn't this use a template? Maybe the real fix is to make core use a template here then we can override it with translatable?

**Comment by tractorcow** [comment]
Easy solution; Don't extend it, override it.

LeftAndMain_TreeNode is instantiated with ::create, which means you can DI it.

Subclass LeftAndMain_TreeNode with a translatable version, then use config to update to the new class.

```yaml
Injector:
  LeftAndMain_TreeNode:
    class: LeftAndMain_TranslatableTreeNode
```

**Comment by tractorcow** [comment]
Oh, and yes please use a template. ;D

**Comment by dhensby** [comment]
Only problem with injector is what it the end user wants to do that too?

**Comment by tractorcow** [comment]
> Only problem with injector is what it the end user wants to do that too?

I don't understand the context of this question. Do you mean how can the end user get a certain behaviour without changing the site code? They can't.

**Comment by JorisDebonnet** [comment]
Haven't had time to try out anything else yet -- but considering using a template, that would of course be a good solution. Not sure which modules would use it apart from Translatable, but I guess that doesn't matter. 

But if we do use a template, do we even need any of the ... stuff you guys are talking about? :D I imagine Framework would get a new template, Translatable creates a template with the same name in some strategic folder, and by pure magic it would all just work. No?

About version numbers: I assume it isn't a breaking change for either framework or translatable, even in translatable because the template simply remains unused if an older framework is used. If it does need an Extension, I guess that one needs a check for the implementation in framework.

**Comment by tractorcow** [comment]
> But if we do use a template, do we even need any of the ... stuff you guys are talking about?

Not really... But if you use injector you can already get the effect you need without any core changes at all.

**Comment by dhensby** [comment]
I think what we're saying is:

1. Use a template and PR that
2. You can use injector to achieve this already if you need it right now.

File: code/controller/TranslatableCMSTreeNodeExtension.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * @package translatable
+ */
+class TranslatableCMSTreeNodeExtension extends Extension {
+
+	function init() {
+		die("Init TranslatableCMSTreeNodeExtension. "); // this never gets called in the current config
```
Comment (tractorcow) [inline code/controller/TranslatableCMSTreeNodeExtension.php]: If it's never called, do we need the die then?

Comment (JorisDebonnet) [inline code/controller/TranslatableCMSTreeNodeExtension.php]: This is just a debug line in an unfinished PR: I did in fact want that line to be reached, but it never did and I couldn't yet figure out why. 

Due to this only being a "nice to have", I haven't yet been able to spend more time on it, but I still intend to later!..

Comment (tractorcow) [inline code/controller/TranslatableCMSTreeNodeExtension.php]: ok, well ping me when you are finished with this PR. :)

**Comment by JorisDebonnet** [comment]
This is totally stale anyway; I'll see whether I get to this again in the future. Closing for now.

**Pull request closed without merging**